README.md

```diff
@@ -379,6 +379,34 @@ Succeeded   : 50
+  - Given that Nextflow supports arbitrary Groovy/Java code, we could technically accomplish all of the work in Nextflow alone. However, it makes a lot more sense to use a programming language that our team (_i.e._ DPE) is familiar with and that people outside of our team can readily contribute to. Python fits the bill. Hence, we are currently building a Python-Nextflow hybrid tool. 
+  - In Nextflow, the `container` directive and the `script` block can leverage values provided as inputs (_e.g._ `container "${some_input}"`). This functionality allows us to keep the Nextflow workflow very general. Otherwise, we would have to hard-code the containers and commands in the Nextflow workflow and somehow keep everything in sync with the Python package, which takes care of interpreting the commands.
+  - The Nextflow workflow makes use of a short `parseJson()` Groovy function to parse the contents of a JSON file into a Nextflow channel. This helps bridge the gap between Python-based processes and Nextflow since Nextflow cannot easily handle multiple outputs that are not files (_e.g._ stdout), especially when the output is intended to be structured.
+- **Python Package**
+  - I use the term "QC target" to refer to any combination of files and folders that represent a type of data and can be validated as a group. As a starting point, we are focusing on single files, so "QC target" and "QC file" can be used interchangeably. However, we will quickly run into multi-file QC targets, such as FASTQ file pairs and BAM/CRAM files with their indices.
+  - Almost every class has a `to_dict()` method to generate a dictionary representation of an instance of that class. This is used primarily to serialize the objects to JSON between steps in the Nextflow workflow. Ideally, the Python package can deserialize these JSON files into the appropriate set of objects.
+  - In the current version, I distinguish between internal and external tests. This is used to separate tests into two branches in the Nextflow DAG.
+  - The QC suite is intended to represent the set of tests that are relevant to each file type. This approach might need to be re-engineered a bit to accommodate multi-file QC targets though. I'm also using inheritance to accumulate additional QC tests. For example, the OME-TIFF QC suite has OME-specific tests in addition to those inherited from the TIFF QC suite, which in turn inherits from the generic file QC suite (with basic tests like MD5 checksum validation).
+
+### Caveats
+
+- The pipeline currently uses container images on Docker Hub, which might be affected by rate limits.
```

Comment:
We can have a discussion as a group about Docker Hub and its future. I wrote a [document](https://docs.google.com/document/d/1GGXITF2k75VWLejpUcQ_7uSsqyN2l0mS3DPZQ2VlXu4/edit) about potentially shifting to GHCR, but there are unknowns about GitHub's application usage costs.